src/lib/list.c

```diff
@@ -318,3 +318,28 @@ int list_strcmp(const void *a, const void *b)
 {
 	return strcmp((const char *)a, (const char *)b);
 }
+
+void list_str_deduplicate(struct list *strs)
```

Comment:
what do you mean is not generic? this is a generic list function, that is why I added it to list.c. Or do you mean is not generic because it is specific for string data types? would you prefer it if it was `list_deduplicate(struct list *list, comparison_fn_type cmd_fn)` instead of `list_str_deduplicate(struct list *list)`?

Reply:
list_deduplicate(struct list *list, comparison_fn_type cmd_fn) would be better so we can use it do deduplicate lists of other elements too, like files and manifests.

Reply:
but it's fine to have a str_deduplicate and implement that when needed

Reply:
ok, I'll make the function generic of data types then 👍

Reply:
done